experiment/measurer/measure_manager.py

```diff
@@ -0,0 +1,327 @@
+
+logger = logs.Logger('measure_manager')  # pylint: disable=invalid-name
+
+FAIL_WAIT_SECONDS = 30
+POLL_RESULTS_WAIT_SECONDS = 5
+SNAPSHOTS_BATCH_SAVE_SIZE = 100
+
+
+def get_experiment_folders_dir():
+    """Return experiment folders directory."""
+    return exp_path.path('experiment-folders')
+
+
+def remote_dir_exists(directory: pathlib.Path) -> bool:
+    """Does |directory| exist in the CLOUD_EXPERIMENT_BUCKET."""
+    return gsutil.ls(exp_path.gcs(directory), must_exist=False)[0] == 0
+
+
+def measure_loop(experiment: str, max_total_time: int, redis_host: str):
+    """Continuously measure trials for |experiment|."""
+    db_utils.initialize()
+    logs.initialize(default_extras={
+        'component': 'dispatcher',
+        'subcomponent': 'measurer',
+    })
+    with multiprocessing.Pool() as pool:
+        set_up_coverage_binaries(pool, experiment)
+    # Using Multiprocessing.Queue will fail with a complaint about
+    # inheriting queue.
+    redis_connection = redis.Redis(host=redis_host)
+    queue = rq.Queue(connection=redis_connection)
+    while True:
+        try:
+            # Get whether all trials have ended before we measure to prevent
+            # races.
+            all_trials_ended = scheduler.all_trials_ended(experiment)
+            if not measure_all_trials(experiment, max_total_time, queue):
+                # We didn't measure any trials.
+                if all_trials_ended:
+                    # There are no trials producing snapshots to measure.
+                    # Given that we couldn't measure any snapshots, we won't
+                    # be able to measure any the future, so break now.
+                    break
+        except Exception:  # pylint: disable=broad-except
+            logger.error('Error occurred during measuring.')
+
+            time.sleep(FAIL_WAIT_SECONDS)
+
+    logger.info('Finished measuring.')
+
+
+def get_job_timeout():
+    """Returns the timeout for an rq job."""
+    return experiment_utils.get_snapshot_seconds() + 2 * 60
+
+
+def measure_all_trials(experiment: str, max_total_time: int, queue) -> bool:
+    """Get coverage data (with coverage runs) for all active trials. Note that
+    this should not be called unless multiprocessing.set_start_method('spawn')
+    was called first. Otherwise it will use fork which breaks logging."""
+    logger.info('Measuring all trials.')
+
+    experiment_folders_dir = get_experiment_folders_dir()
+    if not remote_dir_exists(experiment_folders_dir):
+        return True
+
+    max_cycle = _time_to_cycle(max_total_time)
+    unmeasured_snapshots = get_unmeasured_snapshots(experiment, max_cycle)
+
+    if not unmeasured_snapshots:
+        return False
+
+    job_timeout = get_job_timeout()
+    results = [
+        queue.enqueue(measure_worker.measure_trial_coverage,
+                      unmeasured_snapshot,
+                      job_timeout=job_timeout)
+        for unmeasured_snapshot in unmeasured_snapshots
```

Comment:
Currently we won't ever create a task for measuring snapshot. I'll look into using `depends_on` because it seems more elegant than my current approach. But I need to make sure it works well with with preemptibles where somtimes we won't want to measure snapshot N+1 just because N was measured.

Reply:
> Currently we won't ever create a task for measuring snapshot. 

Oh, i just realized that we enqueue `measure_worker.measure_trial_coverage` at this point! 

>I'll look into using `depends_on` because it seems more elegant than my current approach. 

SG, thanks!

Reply:
Sorry, I left out an important part of this sentence, we won't create a task for measuring snapshot N+1 if N hasn't been measured.